src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const dataOfState = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => {
+          delete stateCopy[key];
+        });
+        break;
+
+      case 'clear':
+        Object.keys(stateCopy).forEach(key => {
+          delete stateCopy[key];
+        });
+        break;
+
+      default:
+        return 'incorect data';
```

Comment:
Even better would be to [throw an error](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Error#%D0%BF%D1%80%D0%B8%D0%BC%D0%B5%D1%80_%D0%B2%D1%8B%D0%B1%D1%80%D0%B0%D1%81%D1%8B%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5_%D0%BE%D0%B1%D1%8B%D1%87%D0%BD%D0%BE%D0%B9_%D0%BE%D1%88%D0%B8%D0%B1%D0%BA%D0%B8), not to return a message